src/infer_response.h

```diff
@@ -94,6 +98,11 @@ class InferenceResponseFactory {
   void ReleaseTrace() { trace_ = nullptr; }
 #endif  // TRITON_ENABLE_TRACING
 
+#ifdef TRITON_ENABLE_STATS
+  // Return the current response statistics index and increment it.
+  uint64_t ResponseStatsIndex() { return response_stats_index_++; };
```

Comment:
Could you briefly explain the logic behind why we need to increment it here?

Reply:
We want to increment to the next index, atomically. I think we can adjust the function naming if it makes it better?

Reply:
Yes I agree with the name adjustment, my concern is that as is `ResponseStatsIndex` implies that the function just returns index. But with the current logic we increment it as well.

Reply:
cc @Tabrizian in case he has a feedback

Reply:
Agreed with Olga. Perhaps we can rename it to `GetAndIncrementResponseIndex` to make it more clear.

Reply:
Sure. Renamed to `GetAndIncrementResponseIndex`.